mods/ra/maps/soviet-04a/main.lua

```diff
@@ -0,0 +1,159 @@
+CheckForRadar = function()
+	Radar = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(actor)
+		return actor.Type == "dome" and actor.Owner == Greece
+	end)
+
+	return #Radar >= 1
+end
+
+CheckForCY = function()
+	cyard = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(actor)
+		return actor.Type == "fact" and actor.Owner == Greece
+	end)
+
+	return #cyard >= 1
+end
+
+RunInitialActivities = function()
+	Harvester.FindResources()
+	IdlingUnits()
+	Trigger.AfterDelay(10, function()
+		BringPatrol1()
+		BringPatrol2()
+		BuildBase()
+
+		local buildings = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(self) return self.Owner == Greece and self.HasProperty("StartBuildingRepairs") end)
+		Utils.Do(buildings, function(actor)
+			Trigger.OnDamaged(actor, function(building)
+				if building.Owner == Greece and building.Health < building.MaxHealth * 3/4 then
+					building.StartBuildingRepairs()
+				end
+			end)
+		end)
+	end)
+
+	Reinforcements.Reinforce(player, SovietMCV, SovietStartToBasePath, 0, function(mcv)
+		mcv.Move(StartCamPoint.Location)
+	end)
+	Media.PlaySpeechNotification(player, "ReinforcementsArrived")
+
+	Trigger.OnKilled(Barr, function(building)
+		BaseBuildings[2][4] = false
+	end)
+
+	Trigger.OnKilled(Proc, function(building)
+		BaseBuildings[3][4] = false
+	end)
+	
+	Trigger.OnKilled(Weap, function(building)
+		BaseBuildings[4][4] = false
+	end)
+
+	Trigger.OnEnteredFootprint(VillageCamArea, function(actor, id)
+		if actor.Owner == player then
+			Actor.Create("camera", true, { Owner = player, Location = VillagePoint.Location })
```

Comment:
Could we have this camera removed once all visible civilians and civilian buildings are destroyed?